### PR TITLE
Store training schema in with_training branch

### PR DESCRIPTION
**Pull request by sidhulyalkar**
- Save training code implementation that can be revisited when we think about implementing a widget to allow users to label image data to provide a ground truth.

### CODE REVIEW

**Comment by ttngu207** [comment]
We can keep it on your `with_training` branch for now, no need to PR to upstream
